File: Basic_Chat_Flutter/chat_app/lib/model/message.dart
class Message {
  String message;
  String sentByMe;

  Message({required this.message, required this.sentByMe});

factory Message.fromJson(Map<String, dynamic> json) {
  return Message(
    message: json["message"] ?? "", // Provide a default value if message is null
    sentByMe: json["sentByMe"] ?? "", // Provide a default value if sentByMe is null
  );
}
}

File: Basic_Chat/entities/message.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const messageSchema = new Schema({
  message: {
    type: String,
    required: true
  },
  sentByMe: {
    type: String,
    required: true
  }
});

// No need to define a constructor in Mongoose schemas

// Static method for creating a Message from JSON data
messageSchema.statics.fromJson = function(json) {
  return new this({
    message: json["message"] || "", // Use || operator for default value
    sentByMe: json["sentByMe"] || ""  // Default to false
  });
};

module.exports = mongoose.model('Message', messageSchema);
